crates/repository/src/lib.rs

```diff
@@ -304,13 +320,28 @@ impl Repository {
         Ok(())
     }
 
-    pub async fn install_sdk(
+    pub async fn link_and_install_sdk(
         &self,
         version: Option<&str>,
         installation_directory: Option<&Path>,
     ) -> Result<()> {
+        let installation_directory = self
+            .link_sdk_home(installation_directory)
+            .await
+            .wrap_err("failed to link SDK home")?;
+
+        let use_docker = OS_IS_NOT_LINUX;
+        if !use_docker {
```

Comment:
Please move this to the call site. The caller should decide what to do. That is, the "normal" install caller now calls two functions, the docker way is only calling the link function. Currently, this function hides a lot of semantic not visible to the caller in any way.

Reply:
I decided against this to avoide the inherent code duplication, should I still inline this?